x/evm/types/state_object.go

```diff
@@ -87,6 +88,7 @@ func newStateObject(db *CommitStateDB, accProto authexported.Account) *stateObje
 	return &stateObject{
 		stateDB:       db,
 		account:       ethermintAccount,
+		balance:       balance,
```

Comment:
Indicating here also just for reference, I don't think this should be necessary, why would you not get the balance from the EthAccount type through the base account type?

Reply:
balances are now decoupled from the account. And because the state object doesn't hold a reference of the bank keeper I added it here